Automattic/woocommerce-payments
# Load Subscriptions Core and WCPay Subscriptions support classes behind a feature flag

Fixes #2987 

#### Changes proposed in this Pull Request





With this PR, Subscriptions Core and WCPay Subscription will only be loaded when the feature flag `_wcpay_feature_subscriptions` is set in wp_options to `1`.

As part of our teams scope call last week, we discussed doing a soft-launch to only US based stores, however, as part of merging `feature/stripe_billing` into `develop` we decided to remove the US-only check and make it only check for the wp-option first.

~When I first added the new `WC_Payments_Features::is_wcpay_subscriptions_enabled()` function, I wrote it with the intention of having the WCPay Subscriptions US-only so I've commented out the code to be used later when we release this.~

**Chris update:** I have removed the commented out code, we can always add it back later when needed.





#### Testing instructions



1. Make sure you have Subscriptions extension deactivated
2. Checkout this `feature/stripe_billing`
3. Confirm WCPay Subscriptions is loaded (check **WooCommerce > Subscriptions**)
4. Checkout this branch
5. Without the feature flag set, confirm that no Subscriptions functionality exists (I just checked there's no Subscriptions settings or Subscriptions table)
6. Manually add `_wcpay_feature_subscriptions` in your wp_options table and set it to `1`
7. Confirm Subscriptions Core is loading (check Subscriptions Settings and table exists)
8. Set `_wcpay_feature_subscriptions` to `0` and check no subscriptions functionality is loaded (opposite of above checks)



-------------------

Something I'm interested in hearing others thoughts on is do we actually need surround the following code in a feature flag check:

```
include_once WCPAY_ABSPATH . '/includes/subscriptions/class-wc-payments-subscriptions.php';
WC_Payments_Subscriptions::init( self::$api_client, self::$customer_service );
```

On its own, I don't think most of those classes that get setup as part of this `init()` don't actually do anything.

Closes #2987
